tooling/BinRepl.java

```diff
@@ -0,0 +1,227 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing,
+ * software # distributed under the License is distributed on an
+ * "AS IS" BASIS, # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package temurin.tools;
+
+import java.io.FileOutputStream;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.HexFormat;
+
+/**
+ * This utility class binary replaces the given "hex" binary values with a new value.
+ * Params:
+ *   --inFile <input>
+ *   --outFile <output>
+ *   --hex "aa:aa-bb:bb" Replace aa:aa with bb:bb
+ *   --string "string1=string2" Replace string1 with string2
+ *   --pad <hh> Pad string replace with given hex byte value
+ *   --firstOnly Only replace first instance
+ *    --32bitBoundaryOnly Only replace if found on a 32bit boundary
+ */
+class BinRepl {
+
+  protected BinRepl() { }
+
+  // A simple static counter
+  private static int replCounter = 0;
+
+  public static void main(final String[] args) throws Exception {
+    String inFile = null;
+    String outFile = null;
+    boolean firstOnly = false;
+    boolean boundary32bitOnly = false;
+
+    String[] hex = null;
+    String[] str = null;
+    String strPad = null;
+
+    for (int i = 0; i < args.length; i++) {
```

Comment:
There's an Apache lib ( https://commons.apache.org/proper/commons-cli/) that can deal with all of this CLI processing...

Reply:
This is a small utility script, I don't want to add a dependency jar, and the parsing is quite trivial. So will leave as is.